### DISCUSSION TITLE
Feature request - paper async chunks compatibility

### DISCUSSION THREAD

**Discussion by ArtNRG** [Zrips/CMI#1288] [Feature Requests]
The 1.13.2 paper has its own async chunk loading feature. It's possible to preload or just load chunks async during any teleports like home, warp, back etc. 
Additional info about the paper is here: https://github.com/PaperMC/Paper/pull/1397

**Comment by TheWranger** [comment]
Are you sure extra compatibility is needed for this? Pretty sure paper automatically makes all chunk loading async with that option.

**Comment by Zrips** [comment]
Hmm in general this chunk async load should be done by server itself, as it should know to load atleast one chunk before teleporting player to avoid glitching throw a blocks and then it could load them in async without any extra code from 3rd party plugins instructing to do so. As this is one of main things which causes server lag and should be enabled by default if server supports async chunk loading.

**Comment by 0-x-2-2** [comment]
[PaperLib](https://github.com/PaperMC/PaperLib) has a api for getting chunks async and teleporting players async with a fallback for spigot
